service/src/main/java/bio/terra/cbas/util/BackfillGithubMethodDetails.java

```diff
@@ -0,0 +1,128 @@
+        PreparedStatement methodVersionStmt = connection.prepareStatement(methodVersionQuery);
+        methodVersionStmt.setObject(1, UUID.fromString(methodId));
+
+        ResultSet methodVersionResultSet = methodVersionStmt.executeQuery();
+
+        if (methodVersionResultSet.next()) {
+          // extract GitHub details for each method
+          String rawUrl = convertGithubToRawUrl(methodVersionResultSet.getString(2));
+          GithubUrlComponents githubMethodDetails = extractGithubDetailsFromUrl(rawUrl);
+
+          // create insert statement for inserting data in 'github_method_details' table
+          InsertStatement insertGithubDetailsStmt =
+              new InsertStatement(
+                      database.getLiquibaseCatalogName(),
+                      database.getLiquibaseSchemaName(),
+                      "github_method_details")
+                  .addColumnValue("repository", githubMethodDetails.repo())
+                  .addColumnValue("organization", githubMethodDetails.org())
+                  .addColumnValue("path", githubMethodDetails.path())
+                  .addColumnValue(
+                      "private",
+                      false) // this is false since all methods until now should have been public
+                  .addColumnValue("method_id", methodId);
+
+          // create update statement for inserting value into 'branch_or_tag_name'column in
+          // 'method_version' table
+          UpdateStatement updateBranchTagStmt =
+              new UpdateStatement(
+                      database.getLiquibaseCatalogName(),
+                      database.getLiquibaseSchemaName(),
+                      "method_version")
+                  .addNewColumnValue("branch_or_tag_name", githubMethodDetails.branchOrTag())
+                  .setWhereClause(
+                      database.escapeObjectName("method_version_id", LiquibaseColumn.class)
+                          + " = ? ")
+                  .addWhereParameters(methodVersionResultSet.getString(1));
+
+          sqlStatementList.add(insertGithubDetailsStmt);
+          sqlStatementList.add(updateBranchTagStmt);
+        } else {
+          // it's an error case if method doesn't have a method url
+          throw new DatabaseException(
+              "No method url found for the method ID '%s'".formatted(methodId));
+        }
+
+        methodVersionStmt.close();
+      }
+
+      return sqlStatementList.toArray(sqlStatementArray);
+    } catch (Exception e) {
+      throw new CustomChangeException(e);
+    }
+  }
+
+  @Override
+  public String getConfirmationMessage() {
+    return "Successfully back filled data (if any) to 'github_method_details' table.";
+  }
+
+  @Override
+  public void setUp() throws SetupException {
+    // do nothing since no initial set up is needed
+  }
+
+  @Override
+  public ValidationErrors validate(Database database) {
+    return null;
```

Comment:
just curious if you considered putting something here, and decided against it, or if the rest of the code is just enough to give you confidence to skip it?

Reply:
I did briefly look into it but wasn't sure what validation we should do since this is essentially being called via the changeset. I guess one validation that can be done is to check if tables used in the backfill `method`, `method_version` and `github_method_details` exist but I wasn't sure if that would make sense to validate since they should have been added before running the backfill changeset. Do you have any suggestions on what should be validated here @mspector ?

Reply:
Thinking out loud for a moment... I assume this runs after the changeset is applied, and if it fails, it rolls back and cancels the migration or something like that.

I think any suggestion I could come up with would be covered by the unit test that you already have, so I wouldn't insist on something being added here.

Reply:
Actually I might have misunderstood the validate method here. Based on the docs it seems to be validating the change before executing it and I wonder if this is needed in our case? 🤔 We are extracting Github details through a method and I am not sure how we would validate various data we are trying to insert to our database dynamically. This might have been useful if we were inserting static data.
>    /**
     * Tests that the change is configured correctly before attempting to execute it.
     * @param database The database the change will be ran against
     */
    ValidationErrors validate(Database database);

Reply:
Oh I see -- I misunderstood too. I agree that we don't need this.